Source/Scene/Cesium3DTileset.js

```diff
@@ -1253,14 +1253,15 @@ define([
 
         var statistics = this._statistics;
 
-        // Append the tileset version to the resource
-        if (!defined(resource.queryParameters.v)) {
-            var versionQuery = {
-                v : defaultValue(asset.tilesetVersion, '0.0')
-            };
-            this._basePath += '?v=' + versionQuery.v;
-            resource.setQueryParameters(versionQuery);
+        var versionQuery = {};
+        if (defined(asset.tilesetVersion)) {
+            // Append the tileset version to the resource
+            versionQuery.v = asset.tilesetVersion;
+            this._basePath += '?v=' + asset.tilesetVersion;
         }
+        // Always set query parameters to make sure the tilesetVersion matches the current tileset
+        // Otherwise a child tileset might use the tilesetVersion from the parent
+        resource.setQueryParameters(versionQuery);
```

Comment:
I literally put an inline comment above this line explaining that

Reply:
Wait, I'm supposed to read those things? :trollface: 

Unfortunately, there is still an issue because the code doesn't actually remove the parameter.  `setQueryParameters` only overwrites values if they are supplied, since `versionQuery` is an empty object, it's essentially a no-op.  

```js
//Assume a query parameter
var url1 = new Cesium.Resource('https://cesium.com?v=1');
console.log(url1.url); // https://cesium.com?v=1

//This is basically a no-op
url1.setQueryParameters({});
console.log(url1.url); // https://cesium.com?v=1

//However, this only clears the value, not the parameter
url1.setQueryParameters({ v: undefined});
console.log(url1.url); // https://cesium.com?v

//To actually delete the query parameter
delete url1.queryParameters.v;
console.log(url1.url); // https://cesium.com
```

So you need an else that deletes it and you can get rid of the tmp object.

Reply:
And sorry for not reading your comment.

Reply:
Ah, yeah I was looking at how `setQueryParamaters` works and for some reason I thought that it would remove the value unless you set the `useAsDefault` argument, but you're right.